provider/aws/aws.go

```diff
@@ -405,6 +400,47 @@ func (p *AWSProvider) doRecords(ctx context.Context, action string, endpoints []
 	return p.submitChanges(ctx, p.newChanges(action, endpoints, records, zones), zones)
 }
 
+// UpdateRecords updates a given set of old records to a new set of records in a given hosted zone.
+func (p *AWSProvider) UpdateRecords(ctx context.Context, updates, current []*endpoint.Endpoint) error {
+	zones, err := p.Zones(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to list zones, aborting UpdateRecords")
+	}
+
+	records, err := p.records(ctx, zones)
+	if err != nil {
+		log.Errorf("failed to list records while preparing UpdateRecords: %s", err)
+	}
+
+	return p.submitChanges(ctx, p.createUpdateChanges(updates, current, records, zones), zones)
+}
```

Comment:
Isn't this exactly the same as `doRecords`? If so, why are we duplicating it?

Reply:
It's not quite the same, since we need to call createUpdateChanges to handle the special case of ALIAS records. The duplication is just getting the zones and records. I suppose I could remove the duplication by passing a function as an argument to doRecords.

Reply:
Oh right, I missed that. I think we can keep it as is then.